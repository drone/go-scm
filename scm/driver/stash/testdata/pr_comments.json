{
    "size": 3,
    "limit": 25,
    "isLastPage": true,
    "values": [
        {
            "id": 14,
            "createdDate": 1530770330632,
            "user": {
                "name": "jcitizen",
                "emailAddress": "jane@example.com",
                "id": 1,
                "displayName": "Jane Citizen",
                "active": true,
                "slug": "jcitizen",
                "type": "NORMAL",
                "links": {
                    "self": [
                        {
                            "href": "http://example.com:7990/users/jcitizen"
                        }
                    ]
                }
            },
            "action": "COMMENTED",
            "commentAction": "ADDED",
            "comment": {
                "properties": {
                    "repositoryId": 1
                },
                "id": 2,
                "version": 0,
                "text": "this is second comment",
                "author": {
                    "name": "jcitizen",
                    "emailAddress": "jane@example.com",
                    "id": 1,
                    "displayName": "Jane Citizen",
                    "active": true,
                    "slug": "jcitizen",
                    "type": "NORMAL",
                    "links": {
                        "self": [
                            {
                                "href": "http://example.com:7990/users/jcitizen"
                            }
                        ]
                    }
                },
                "createdDate": 1530770330632,
                "updatedDate": 1530770330632,
                "comments": [{
                    "properties": {
                        "repositoryId": 1
                    },
                    "id": 3,
                    "version": 0,
                    "text": "this is reply to second comment",
                    "author": {
                        "name": "jcitizen",
                        "emailAddress": "jane@example.com",
                        "id": 1,
                        "displayName": "Jane Citizen",
                        "active": true,
                        "slug": "jcitizen",
                        "type": "NORMAL",
                        "links": {
                            "self": [
                                {
                                    "href": "http://example.com:7990/users/jcitizen"
                                }
                            ]
                        }
                    },
                    "createdDate": 1530770345640,
                    "updatedDate": 1530770345640,
                    "comments": [],
                    "tasks": [],
                    "permittedOperations": {
                        "editable": true,
                        "deletable": true
                    }
                }],
                "tasks": [],
                "permittedOperations": {
                    "editable": true,
                    "deletable": true
                }
            }
        },
        {
            "id": 13,
            "createdDate": 1530770325043,
            "user": {
                "name": "jcitizen",
                "emailAddress": "jane@example.com",
                "id": 1,
                "displayName": "Jane Citizen",
                "active": true,
                "slug": "jcitizen",
                "type": "NORMAL",
                "links": {
                    "self": [
                        {
                            "href": "http://example.com:7990/users/jcitizen"
                        }
                    ]
                }
            },
            "action": "COMMENTED",
            "commentAction": "ADDED",
            "comment": {
                "properties": {
                    "repositoryId": 1
                },
                "id": 1,
                "version": 0,
                "text": "this is a comment",
                "author": {
                    "name": "jcitizen",
                    "emailAddress": "jane@example.com",
                    "id": 1,
                    "displayName": "Jane Citizen",
                    "active": true,
                    "slug": "jcitizen",
                    "type": "NORMAL",
                    "links": {
                        "self": [
                            {
                                "href": "http://example.com:7990/users/jcitizen"
                            }
                        ]
                    }
                },
                "createdDate": 1530770325043,
                "updatedDate": 1530770325043,
                "comments": [],
                "tasks": [],
                "permittedOperations": {
                    "editable": true,
                    "deletable": true
                }
            }
        },
        {
            "id": 12,
            "createdDate": 1530766871157,
            "user": {
                "name": "jcitizen",
                "emailAddress": "jane@example.com",
                "id": 1,
                "displayName": "Jane Citizen",
                "active": true,
                "slug": "jcitizen",
                "type": "NORMAL",
                "links": {
                    "self": [
                        {
                            "href": "http://example.com:7990/users/jcitizen"
                        }
                    ]
                }
            },
            "action": "OPENED"
        }
    ],
    "start": 0
}
