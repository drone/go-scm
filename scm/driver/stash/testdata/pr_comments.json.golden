[
  {
    "ID": 2,
    "Body": "this is second comment",
    "Author": {
      "ID": "",
      "Login": "jcitizen",
      "Name": "Jane Citizen",
      "Email": "jane@example.com",
      "Avatar": "https://www.gravatar.com/avatar/9e26471d35a78862c17e467d87cddedf.jpg",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Created": "2018-07-05T11:28:50+05:30",
    "Updated": "2018-07-05T11:28:50+05:30",
    "ParentID": 0
  },
  {
    "ID": 3,
    "Body": "this is reply to second comment",
    "Author": {
      "ID": "",
      "Login": "jcitizen",
      "Name": "Jane Citizen",
      "Email": "jane@example.com",
      "Avatar": "https://www.gravatar.com/avatar/9e26471d35a78862c17e467d87cddedf.jpg",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Created": "2018-07-05T11:29:05+05:30",
    "Updated": "2018-07-05T11:29:05+05:30",
    "ParentID": 2
  },
  {
    "ID": 1,
    "Body": "this is a comment",
    "Author": {
      "ID": "",
      "Login": "jcitizen",
      "Name": "Jane Citizen",
      "Email": "jane@example.com",
      "Avatar": "https://www.gravatar.com/avatar/9e26471d35a78862c17e467d87cddedf.jpg",
      "Created": "0001-01-01T00:00:00Z",
      "Updated": "0001-01-01T00:00:00Z"
    },
    "Created": "2018-07-05T11:28:45+05:30",
    "Updated": "2018-07-05T11:28:45+05:30",
    "ParentID": 0
  }
]